### ISSUE TITLE
Switch Auth0 with JupyterHub CILogon OAuthenticator

### ISSUE THREAD

**Issue by GeorgianaElena** [2i2c-org/infrastructure#1304]
### Context

As part of https://github.com/2i2c-org/infrastructure/issues/967, we decided and implemented the infrastructure for using [`CILogon`](https://www.cilogon.org/) as the authenticator for the 2i2c hubs.

This was possible thanks to
- https://github.com/jupyterhub/oauthenticator
- the CILogon folks who offered a free trial of their admin API

The JupyterHub CILogon OAuthenticator is currently used by [several hubs](https://github.com/2i2c-org/infrastructure/search?q=CILogonOAuthenticator):
- ANU hub
- 2i2c demo hub
- cloudbank demo hub
- dask-staging

### Proposal

I propose thinking a strategy to migrate all the hubs to using the CILogonOAuthenticator if we decide to do it.
 Also, provide feedback and share our experience with the CILogon folks, possible though a blog post.

### Updates and actions

This needs to happen first I believe

- [x] https://github.com/2i2c-org/meta/issues/317

Also, checkout @choldgraf proposal here for a possible rollout process https://github.com/2i2c-org/infrastructure/issues/967#issuecomment-1088799731

Relevant issues that are worth to be kept an eye on in this context:
- https://github.com/jupyterhub/oauthenticator/issues/515
- https://github.com/jupyterhub/oauthenticator/issues/522
- https://github.com/ncsa/OA4MP/issues/45

**Comment by GeorgianaElena** [comment]
New oathenticator version released. See https://github.com/2i2c-org/infrastructure/issues/1319#issuecomment-1147505706 for details

**Comment by GeorgianaElena** [comment]
I will be closing this issue, since all of the action points have been done. We still need to remove the current setup for Auth0 from our infra, I believe. But we could open a new issue for that

**Comment by GeorgianaElena** [comment]
Actually, this is the issue that actually tracks the removal of Auth0. The tile was a bit confusing. Sorry about that

**Comment by yuvipanda** [comment]
100% I agree we should get rid of auth0 :)
